### PR TITLE
tikv: recheck kv status before invalidate region on sending fail

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve?

Problem Summary:

the simple and cherry-pickable part of https://github.com/pingcap/tidb/pull/15989

do addition kv status check before invalidate region to avoid false-invalidate region cache when tikv isn't down but sending request failure(like meet network timeout).

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

do addition kv status check before invalidate region

How it Works:

do addition kv status check before invalidate region

and add metric to new check operation.

if status API unavaliable tidb will works as code before this pr

### Related changes

- Need to cherry-pick to the release branch 4.0 to help large-txn test

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- n/a

### Release note 

recheck kv status before invalidating region on sending fail

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/16933)

### CODE REVIEW

**Comment by bobotu** [comment]
LGTM

File: config/config.go [hunk 1]
```diff
@@ -451,6 +453,8 @@ type TiKVClient struct {
 	// If a store has been up to the limit, it will return error for successive request to
 	// prevent the store occupying too much token in dispatching level.
 	StoreLimit int64 `toml:"store-limit" json:"store-limit"`
+	// StoreLivenessTimeout in nanosecond is the timeout for store liveness check request.
```
Comment (coocood) [inline config/config.go:456]: The default value is in ms, but this comment is in ns.
How about making this option a duration string for easy to use?

Comment (lysu) [reply config/config.go]: done~

File: config/config.toml.example [hunk 2]
```diff
@@ -347,6 +347,9 @@ region-cache-ttl = 600
 # default 0 means shutting off store limit.
 store-limit = 0
 
+# store-liveness-timeout is used to control timeout for store liveness after sending request failed(unit: ms).
+store-liveness-timeout = "120s"
```
Comment (tiancaiamao) [inline config/config.toml.example:351]: 120s
unit : ms  ?

**Comment by tiancaiamao** [comment]
LGTM @coocood

**Comment by lysu** [comment]
friendly ping @coocood

**Pull request merged**